File: CompetitiveProgramming/DMOJ/BlueBook/BlueBook - Exact Divisors/ExactDivisors.hs
{-
-----------------------------------------------------------------------------------
---------------------------- BlueBook - Exact Divisors ----------------------------
Print all exact divisors of a number in any order. Includes 1 and N.

------------------------------------------------------------------------
---------------------------- SOLUTION LOGIC ----------------------------
Check every number from 1 to N. O(N) with loop.

--------------------------------------------------------------------------
---------------------------- PROGRAM COMMENTS ----------------------------
Program to find all exact divisors of a number.
@author Vincent Zhang
@since 17 August 2021
@context Self-Learning Haskell
-}

{-
Solves the problem.
@return IO
-}
main :: IO ()
main = do
  n <- (read :: String -> Int) <$> getLine
  mapM_ print [x | x <- [1..n], mod n x == 0]

File: CompetitiveProgramming/DMOJ/BlueBook/BlueBook - Exact Divisors/ExactDivisors.java
/*
-----------------------------------------------------------------------------------
---------------------------- BlueBook - Exact Divisors ----------------------------
Print all exact divisors of a number in any order. Includes 1 and N.

------------------------------------------------------------------------
---------------------------- SOLUTION LOGIC ----------------------------
Check every number from 1 to N. O(N) with loop.
*/

import java.util.Scanner;

/**
 * Program to find all exact divisors of a number.
 * @author Vincent Zhang
 * @since 17 August 2021
 */
public class ExactDivisors
{
    /**
     * Solves the problem.
     * @param args Command-line arguments
     * @return void
     */
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.close();

        for(int i=1; i<=n; i++)
        {
            if(n% i == 0)
                System.out.println(i);
        }
    }
}